piccolo/testing/model_builder.py

```diff
@@ -159,29 +151,110 @@ def _randomize_attribute(cls, column: Column) -> t.Any:
             Column class to randomize.
 
         """
-        random_value: t.Any
-        if column.value_type == Decimal:
-            precision, scale = column._meta.params["digits"] or (4, 2)
-            random_value = RandomBuilder.next_float(
-                maximum=10 ** (precision - scale), scale=scale
-            )
-        elif column.value_type == datetime.datetime:
-            tz_aware = getattr(column, "tz_aware", False)
-            random_value = RandomBuilder.next_datetime(tz_aware=tz_aware)
-        elif column.value_type == list:
-            length = RandomBuilder.next_int(maximum=10)
-            base_type = t.cast(Array, column).base_column.value_type
-            random_value = [
-                cls.__DEFAULT_MAPPER[base_type]() for _ in range(length)
-            ]
-        elif column._meta.choices:
+        reg = cls.get_registry(column)
+        if column._meta.choices:
             random_value = RandomBuilder.next_enum(column._meta.choices)
         else:
-            random_value = cls.__DEFAULT_MAPPER[column.value_type]()
+            random_value = reg[column.value_type]()
 
-        if "length" in column._meta.params and isinstance(random_value, str):
-            return random_value[: column._meta.params["length"]]
-        elif isinstance(column, (JSON, JSONB)):
+        if isinstance(column, (JSON, JSONB)):
             return json.dumps({"value": random_value})
-
         return random_value
+
+    @classmethod
+    def get_registry(
+        cls, column: Column
+    ) -> MappingProxyType[t.Type, t.Callable]:
+        """
+        This serves as the public API allowing users to **view**
+        the complete registry for the specified column.
+
+        :param column:
+            Column class to randomize.
+
+        """
+        default_mapper = cls.__DEFAULT_MAPPER
+        if not default_mapper:  # execute once only
+            for typ, callable_ in RandomBuilder.get_mapper().items():
+                default_mapper[typ] = callable_
+
+        # order matters
+        reg = {
+            **default_mapper,
+            **cls._get_local_mapper(column),
+            **cls._get_other_mapper(column),
+        }
+
+        if column.value_type == list:
+            reg[list] = partial(
+                RandomBuilder.next_list,
+                reg[t.cast(Array, column).base_column.value_type],
+            )
+        return MappingProxyType(reg)
+
+    @classmethod
+    def _get_local_mapper(cls, column: Column) -> t.Dict[t.Type, t.Callable]:
+        """
+        This classmethod encapsulates the desired logic, utilizing information
+        from the column.
+
+        :param column:
+            Column class to randomize.
+        """
+        local_mapper: t.Dict[t.Type, t.Callable] = {}
+
+        precision, scale = column._meta.params.get("digits") or (4, 2)
+        local_mapper[Decimal] = partial(
+            RandomBuilder.next_decimal, precision, scale
+        )
```

Comment:
If `RandomBuilder.next_decimal` incorporates default values for precision and scale (like this PR), the logic can be simplified to:
```python
        if precision_scale := column._meta.params.get("digits"):
            local_mapper[Decimal] = partial(
                RandomBuilder.next_decimal, *precision_scale
            )
```